packages/semcom-sdk/lib/addListener.ts

```diff
@@ -1,18 +1,20 @@
+import { ComponentDataTypes } from '@digita-ai/semcom-core';
 import { ComponentAppendEvent, ComponentEventTypes, ComponentReadEvent, ComponentResponseEvent, ComponentWriteEvent } from './component/models/component-events.model';
 
 declare global {
   interface GlobalEventHandlersEventMap {
     [ComponentEventTypes.READ]: ComponentReadEvent;
-    [ComponentEventTypes.WRITE]: ComponentWriteEvent;
-    [ComponentEventTypes.APPEND]: ComponentAppendEvent;
-    [ComponentEventTypes.RESPONSE]: ComponentResponseEvent;
+    [ComponentEventTypes.WRITE]: ComponentWriteEvent<any>;
+    [ComponentEventTypes.APPEND]: ComponentAppendEvent<any>;
+    [ComponentEventTypes.RESPONSE]: ComponentResponseEvent<any>;
   }
 }
 
-export const addListener = <T extends ComponentEventTypes>(
+export const addListener = <D extends keyof ComponentDataTypes, T extends ComponentEventTypes>(
```

Comment:
The problem here is that, even with the nice construction, the application still has to separately add a listener for each datatype.